larpandoracontent/LArTwoDReco/LArClusterAssociation/MvaLowEClusterMergingAlgorithm.cc

```diff
@@ -0,0 +1,750 @@
+                CartesianVector vertexPosition(0.f, 0.f, 0.f);
+
+		vertexPosition = LArGeometryHelper::ProjectPosition(this->GetPandora(), pVertex->GetPosition() , hitType);
+		const CartesianVector vtxToCluster(centroid1.GetX() - vertexPosition.GetX(), 0.f, centroid1.GetZ() - vertexPosition.GetZ());
+                const CartesianVector vtxToOtherCluster(centroid2.GetX() - vertexPosition.GetX(), 0.f, centroid2.GetZ() - vertexPosition.GetZ());
+		const float D1(std::sqrt((vtxToCluster.GetX() * vtxToCluster.GetX()) + (vtxToCluster.GetZ() * vtxToCluster.GetZ())));
+		const float D2(std::sqrt((vtxToOtherCluster.GetX() * vtxToOtherCluster.GetX()) + (vtxToOtherCluster.GetZ() * vtxToOtherCluster.GetZ())));
+	        vtxClusterAngle = (vtxToCluster.GetDotProduct(vtxToOtherCluster) / (D1 * D2));
+	     
+	     	std::cout << "Vertex Cluster Angle: " << vtxClusterAngle << std::endl;
+	    }
+	    std::cout << "Vertex List: " << pVertexList << std::endl;
+	    std::cout << "Centroid Separation is: " << centroidSeparation << std::endl;
+            std::cout << "Centroid Vector is: " << centroidVector << std::endl;
+            std::cout << "Cluster1 pos: " << centroid1 << std::endl;
+	    std::cout << "Cluster2 pos: " << centroid2 << std::endl;
+
+	    this->EdgeHitFinder(otherCluster, otherClusterEdgeHits);
+            // cluster->GetOrderedCaloHitList().FillCaloHitList(clusterEdgeHits);
+	    // otherCluster->GetOrderedCaloHitList().FillCaloHitList(otherClusterEdgeHits);
+
+	    const unsigned int otherClusterNHits{otherCluster->GetNCaloHits()};
+      	    const unsigned long int otherClusterNEdgeHits{otherClusterEdgeHits.size()};
+	    
+	   // std::cout << "Comparing Cluster 1 and 2, Cluster 1 has " << clusterNEdgeHits << " edge hits whilst Cluster 2 has " <<                         otherClusterNEdgeHits << std::endl;
+	    std::cout << "Running 2nd Loop" << std::endl;
+            CaloHitList largestList;
+	    CaloHitList smallestList;
+	    CartesianVector vectorTool(0.f, 0.f, 0.f);
+	    if ( clusterEdgeHits.size() > otherClusterEdgeHits.size() )
+            {
+		    largestList = clusterEdgeHits;
+		    smallestList = otherClusterEdgeHits;
+		    vectorTool = centroid2;
+            }
+	    else 
+            {
+                largestList = otherClusterEdgeHits;
+                smallestList = clusterEdgeHits;
+		vectorTool = centroid1;
+	    }
+
+	    std::vector<float> distanceDistribution;
+            std::vector<double>  angleDistribution;
+	    std::vector<float> nearestHitDistribution;
+	    std::vector<float> nearestHitAngleDistribution;
+
+	    float finalDistance{0};
+            double finalAngle{0};
+            float maxEdgeHitSeparation{0};
+	    float minEdgeHitSeparation{10000};
+            int contact{0};
+	    int proximity{0};
+	    float contactThreshold{2};
+	    float proximityThreshold{5}; 
+	    float adc1(0);
+	    float adc2(0);
+
+	    for (auto iter3 = largestList.begin() ; iter3 != largestList.end() ; ++iter3)
```

Comment:
Looks like you're not using relative iterators here, and for iter4, so better to just use the
for (type variable : list) construct instead